src/Std/Time/Time/Unit/Second.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Internal.Rat
+import Std.Time.Time.Unit.Nanosecond
+
+namespace Std
+namespace Time
+namespace Second
+open Std.Internal
+open Internal
+
+set_option linter.all true
+
+/--
+`Ordinal` represents a bounded value for second, which ranges between 0 and 59 or 60. This accounts
+for potential leap second.
+-/
+def Ordinal (leap : Bool) := Bounded.LE 0 (.ofNat (if leap then 60 else 59))
+
+instance : ToString (Ordinal leap) where
+  toString x := toString x.val
+
+instance : Repr (Ordinal l) where
+  reprPrec r := reprPrec r.val
+
+instance : OfNat (Ordinal leap) n := by
+  have inst := inferInstanceAs (OfNat (Bounded.LE 0 (0 + (59 : Nat))) n)
+  cases leap
+  · exact inst
+  · exact ⟨inst.ofNat.expandTop (by decide)⟩
+
+instance : OfNat (Ordinal true) 60 where
+  ofNat := Bounded.LE.mk (Int.ofNat 60) (by decide)
+
+/--
+`Offset` represents an offset in seconds. It is defined as an `Int`.
+-/
+def Offset : Type := UnitVal 1
```

Comment:
I think it would be cool to make it slightly easier to move between the various `Offset` types and `Duration`.

Right now, to declare a duration of 4 minutes and 4 seconds, I have to do something like
`let duration : Duration := .ofSeconds ((4 : Minute.Offset).toSeconds + (14 : Second.Offset))`. Of course I love the C++ literals, where you can just write `auto duration = 4min + 14s;`, but I think being able to write `let duration : Duration := (4 : Minute.Offset) + (14 : Second.Offset)` using a coercion `Minutes.Offset -> Seconds.Offset` and `Seconds.Offset -> Duration` would already be a significant improvement.

Reply:
What do you think about a macro like `duration(1d 1m 2s)`? I'm going to add some coercions but I think that we could have something like this macro.

Reply:
I like this macro (potentially using the ISO notation for durations), but I would suggest to address this in a future PR so that that the current PR doesn't grow even more.